### PR TITLE
[DEV-2955] bankAdapter

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/shared/models/asset/asset.entity.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export class Asset extends IEntity {
   ikna: boolean;
 
   @Column({ length: 256, default: Blockchain.DEFICHAIN })
-  blockchain: Blockchain;
+  blockchain: Blockchain; //| BankName | ExchangeName;
```
Comment (davidleomay) [inline src/shared/models/asset/asset.entity.ts:73]: What about this? 😉

File: src/subdomains/core/liquidity-management/enums/index.ts [hunk 2]
```diff
@@ -1,14 +1,8 @@
...
-  OPTIMISM = 'Optimism',
-  BINANCE = 'Binance',
-  KRAKEN = 'Kraken',
-  BANK = 'Bank',
-}
+import { Blockchain } from 'src/integration/blockchain/shared/enums/blockchain.enum';
+import { ExchangeName } from 'src/integration/exchange/enums/exchange.enum';
+import { BankName } from 'src/subdomains/supporting/bank/bank/bank.entity';
+
+export type LiquidityManagementType = Blockchain | BankName | ExchangeName;
```
Comment (davidleomay) [inline src/subdomains/core/liquidity-management/enums/index.ts:6]: I would still call it `LiquidityManagementContext`

File: src/subdomains/core/liquidity-management/entities/liquidity-management-rule.entity.ts [hunk 3]
```diff
@@ -14,7 +17,7 @@ import { LiquidityManagementAction } from './liquidity-management-action.entity'
 @Index((r: LiquidityManagementRule) => [r.context, r.targetAsset, r.targetFiat], { unique: true })
 export class LiquidityManagementRule extends IEntity {
   @Column({ length: 256, nullable: true })
-  context: LiquidityManagementContext;
+  context: Blockchain | BankName | ExchangeName;
```
Comment (davidleomay) [inline src/subdomains/core/liquidity-management/entities/liquidity-management-rule.entity.ts:20]: `LiquidityManagementType`?

File: src/subdomains/supporting/bank-tx/bank-tx/services/bank-tx-batch.service.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import { BankTxBatch } from '../entities/bank-tx-batch.entity';
+import { BankTxBatchRepository } from '../repositories/bank-tx-batch.repository';
+
+@Injectable()
+export class BankTxBatchService {
+  constructor(private readonly bankTxBatchRepo: BankTxBatchRepository) {}
+
+  async getBankTxBatchByIban(iban: string): Promise<BankTxBatch> {
+    return this.bankTxBatchRepo.findOneBy({ iban });
+  }
```
Comment (davidleomay) [inline src/subdomains/supporting/bank-tx/bank-tx/services/bank-tx-batch.service.ts:11]: We should probably get the latest?

File: src/subdomains/core/liquidity-management/adapters/balances/bank.adapter.ts [hunk 5]
```diff
@@ -1,21 +1,63 @@
...
+      ),
+    );
+
+    return balances.reduce((prev, curr) => prev.concat(curr), []);
   }
 
-  getNumberOfPendingOrders(_fiat: Fiat): Promise<number> {
+  getNumberOfPendingOrders(_asset: Asset): Promise<number> {
     throw new Error(`Method not implemented.`);
   }
```
Comment (davidleomay) [inline src/subdomains/core/liquidity-management/adapters/balances/bank.adapter.ts:38]: `return 0;`?

File: src/shared/models/asset/asset.entity.ts (hunk 1 above)
Comment (Yannick1712) [reply src/shared/models/asset/asset.entity.ts:73]: Haha definitely needed 😁

File: src/subdomains/supporting/bank-tx/bank-tx/services/bank-tx-batch.service.ts (hunk 4 above)
Comment (Yannick1712) [reply src/subdomains/supporting/bank-tx/bank-tx/services/bank-tx-batch.service.ts:11]: Ah dump, I thought about it before I pushed it but forgot it 😅

**Comment by Yannick1712** [comment]
- [ ] balances from setting service for asset 269, 270, 271

**Review by davidleomay — APPROVED** [review]

**Pull request merged**